packages/opentelemetry-context-async-hooks/src/AsyncHooksContextManager.ts

```diff
@@ -283,6 +236,38 @@ export class AsyncHooksContextManager implements ContextManager {
    * @param uid uid of the async context
    */
   private _destroy(uid: number) {
-    this._contextRefs.delete(uid);
+    this._contexts.delete(uid);
+  }
+
+  /**
+   * Before hook is called just beforing executing a async context.
+   * @param uid uid of the async context
+   */
+  private _before(uid: number) {
+    const context = this._contexts.get(uid);
+    if (context !== undefined) {
+      this._enterContext(context);
+    }
+  }
+
+  /**
+   * After hook is called just after completing the execution of a async context.
+   */
+  private _after() {
+    this._exitContext();
```

Comment:
Would it be better to call `pop` here to avoid the function call overhead of calling `_exitContext`? The duplication is minimal and I think the context manager is one place where performance micro-optimizations are probably warranted.

Reply:
I would prefer not to make it explicit. Also all the cases of `exitContext` and `enterContext` are trivial inline optimization that V8 does automatically anyway, WDYT ?